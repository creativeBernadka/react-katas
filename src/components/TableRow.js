import React, { useEffect, useState } from "react";
import withSizes from "../hocs/withSizes";

const TableRow = ({
  value,
  category,
  subcategory,
  id,
  showEditForm,
  isMobile,
}) => {
  const [showEditButton, setShowEditButton] = useState(true);

  useEffect(() => {
    if (isMobile) {
      setShowEditButton(false);
    } else {
      setShowEditButton(true);
    }
  }, [isMobile]);
  return (
    <tr>
      <td className="spending-details__cell">{value}</td>
      <td className="spending-details__cell">{category}</td>
      <td className="spending-details__cell">{subcategory}</td>
      <td className="spending-details__cell">
        {showEditButton && (
          <button
            className="spending-details__edit-button"
            onClick={() => showEditForm(id)}
          >
            EDIT
          </button>
        )}
      </td>
    </tr>
  );
};

export default withSizes(TableRow);
