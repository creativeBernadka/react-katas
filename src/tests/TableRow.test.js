import React from "react";
import { shallow, mount, render } from "enzyme";
import TableRow from "../components/TableRow";

const defaultProps = {
  value: 12,
  category: "test",
  subcategory: "test",
  id: 1,
  showEditForm: false,
  isMobile: false,
};

describe("<TableRow />", () => {
  const setup = buildSetup(TableRow, defaultProps);

  it("renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("doesn't render button on mobile", () => {
    const wrapper = setup({ isMobile: true });
    expect(wrapper.find(".spending-details__edit-button")).toHaveLength(0);
  });

  it("calls useEffect on isMobile change", () => {
    const spy = jest.spyOn(React, "useEffect").mockImplementation((f) => f());
    // const wrapper = shallow(
    //   <table>
    //     <tbody>
    //       <TableRow {...defaultProps} />
    //     </tbody>
    //   </table>
    // );
    const wrapper = render(<TableRow {...defaultProps} />);
    wrapper.setProps({ isMobile: true });
    wrapper.setProps({ isMobile: false });
    expect(spy).toBeCalled();
    spy.mockClear();
    spy.mockRestore();
  });
});
