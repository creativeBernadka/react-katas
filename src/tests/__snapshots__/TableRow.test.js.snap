// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> renders correctly 1`] = `ShallowWrapper {}`;
